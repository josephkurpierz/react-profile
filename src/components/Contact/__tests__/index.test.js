import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ContactForm from "..";

afterEach(cleanup);

describe('Contact component', () => {
  it('renders', () => {
    render(<ContactForm/>)
  })
  it('matches snapshot', ()=>{
    const { asFragment } = render(<ContactForm/>)
    expect(asFragment()).toMatchSnapshot()
  })
})

describe('h1tag', ()=>{
  it('mtches', () =>{
    render(<ContactForm/>)
    expect(screen.getByTestId('contact')).toHaveTextContent('Contact me')
  })

  describe('submit button', () => {
    render(<ContactForm/>)
    expect(screen.getByTestId('subButt')).toHaveTextContent('Submit')
  })
})