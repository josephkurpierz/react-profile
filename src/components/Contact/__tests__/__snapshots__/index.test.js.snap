// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="contact"
    >
      Contact me
    </h1>
    <form
      id="contact-form"
    >
      <div>
        <label
          for="name"
        >
          Name:
        </label>
        <input
          name="name"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="email"
        >
          email address:
        </label>
        <input
          name="email"
          type="email"
          value=""
        />
      </div>
      <div>
        <label
          for="message"
        >
          message:
        </label>
        <textarea
          name="message"
          rows="5"
        />
      </div>
      <button
        data-testid="subButt"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
